src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,224 @@
-    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo)
+    new ExportEntitiesByTypeActor(app.rawlsDAO, app.googleServicesDAO, userInfo)
 }
 
-class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val userInfo: UserInfo)(implicit protected val executionContext: ExecutionContext) extends Actor with ExportEntitiesByType {
+class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val googleDAO: GoogleServicesDAO, val userInfo: UserInfo)(implicit protected val executionContext: ExecutionContext) extends Actor with ExportEntitiesByType {
+  // Requires its own actor context to work with downstream actors: TSVWriterActor and StreamingActor
+  def actorRefFactory: ActorContext = context
   override def receive: Receive = {
-    case ExportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) => exportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) pipeTo sender
+    case ExportEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) => streamEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) pipeTo sender
   }
 }
 
 trait ExportEntitiesByType extends FireCloudRequestBuilding {
   val rawlsDAO: RawlsDAO
+  val googleDAO: GoogleServicesDAO
   implicit val userInfo: UserInfo
   implicit protected val executionContext: ExecutionContext
+  implicit def actorRefFactory: ActorRefFactory
 
-  def exportEntities(workspaceNamespace: String, workspaceName: String, filename: String, entityType: String, attributeNames: Option[IndexedSeq[String]]): Future[PerRequestMessage] = {
-    rawlsDAO.fetchAllEntitiesOfType(workspaceNamespace, workspaceName, entityType) map { entities =>
-      ModelSchema.getCollectionMemberType(entityType) match {
-        case Success(Some(collectionType)) =>
-          val collectionMemberType = ModelSchema.getPlural(collectionType)
-          val entityData = TSVFormatter.makeEntityTsvString(entities, entityType, attributeNames)
-          val membershipData = TSVFormatter.makeMembershipTsvString(entities, entityType, collectionType, collectionMemberType.get)
-          val zipBytes: Array[Byte] = getZipBytes(entityType, membershipData, entityData)
-          val zippedFileName = entityType + ".zip"
-          RequestCompleteWithHeaders(
-            HttpEntity(ContentTypes.`application/octet-stream`, zipBytes),
-            HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> zippedFileName)))
-        case _ =>
-          val data = TSVFormatter.makeEntityTsvString(entities, entityType, attributeNames)
-          RequestCompleteWithHeaders(
-            (OK, data),
-            HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> filename)),
-            HttpHeaders.`Content-Type`(`text/plain`))
+  implicit val timeout = Timeout(1 minute)
+
+  lazy val logger: Logger = LoggerFactory.getLogger(getClass)
```

Comment:
I was actually confused by what we should be doing. In some cases we do both, `LazyLogging` and `LoggerFactory`. Will switch to `LazyLogging`